pkg/manifests/config_test.go

```diff
@@ -234,3 +235,79 @@ func TestHttpProxyConfig(t *testing.T) {
 		}
 	}
 }
+
+type fakePodsCapacity int
+
+func (fpc *fakePodsCapacity) PodsCapacity(context.Context) (int, error) {
+	return int(*fpc), nil
+}
+func TestEnforceBodySizeLimit(t *testing.T) {
+
+	mc_10 := fakePodsCapacity(10)
+	mc_1000 := fakePodsCapacity(1000)
+	for _, tt := range []struct {
+		name                string
+		config              string
+		expectBodySizeLimit string
+		expectError         bool
+		mockClient          *fakePodsCapacity
+	}{
+		{
+			name:                "empty config",
+			config:              "",
+			expectBodySizeLimit: "2048KB",
+			expectError:         false,
+			mockClient:          &mc_10,
+		},
+		{
+			name:                "disable body size limit",
+			config:              `{"prometheusK8s": {"enforcedBodySizeLimit": "0"}}`,
+			expectBodySizeLimit: "0",
+			expectError:         false,
+			mockClient:          &mc_10,
+		},
+		{
+			name:                "normal size format",
+			config:              `{"prometheusK8s": {"enforcedBodySizeLimit": "10KB"}}`,
+			expectBodySizeLimit: "10KB",
+			expectError:         false,
+			mockClient:          &mc_10,
+		},
+		{
+			name:                "automatically deduced limit: minimal 2MB",
+			config:              `{"prometheusK8s": {"enforcedBodySizeLimit":  ""}}`,
+			expectBodySizeLimit: "2048KB",
+			expectError:         false,
+			mockClient:          &mc_10,
+		},
+		{
+			name:                "automatically deduced limit: larger than minimal 2MB",
+			config:              `{"prometheusK8s": {"enforcedBodySizeLimit": ""}}`,
+			expectBodySizeLimit: "9765KB",
+			expectError:         false,
+			mockClient:          &mc_1000,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			c, err := NewConfigFromString(tt.config)
+			if tt.expectError && err == nil {
+				t.Fatalf("expected error, got nil")
+			} else if tt.expectError && err != nil {
+				return // expected error, got error, skip size verification
+			} else if !tt.expectError && err != nil {
+				t.Fatalf("expected no error, got error %v", err)
+			}
```

Comment:
IIUC NewConfigFromString() should never fail.

```suggestion
			if  err != nil {
				t.Fatalf("expected no error, got error %v", err)
			}
```

Reply:
agree, these checks should be done for LoadEnforcedBodySizeLimit()